### DISCUSSION TITLE
Become independent of mongodb

### DISCUSSION THREAD

**Discussion by jankapunkt** [meteor/meteor#12058] [Feature Requests]
I wonder if Meteor as a framework wants to emancipate from the tight mongodb integration. From my pov this can be implemented in a way, that 

- all current core packages with tight mongo coupling could use adapters, that implement a certain db access scheme
- additional packages using mongo implementations for the adapters would be available out-of-the-box (so we still have the zero-config setup that just runs :smiley: )
- devs can implement their own adapters so they can have a 100% mongo-independent setup and/or provide them as atmosphere packages

Is this something that people want to see (beside me)?

**Comment by paulincai** [comment]
Hi, I personally don't have an imediate need for this but I am wondering anyway what would be the motivation behind it.
I do understand that Meteor could work with other DBs (and it does) but I don't really see the emancipation factor in decoupling from Mongo. You could just build your DB API around another DB or run a NODE in a different flavor.

Is querying a problem? Is it that you would prefer a more relational DB or are you thinking more to opening to another NOSQL DB (e.g. AWS Dynamo). My thinking is if a client needs something specific other than Mongo, I just use a different framework. I am really curious to see your motivation/interest.

**Reply by jankapunkt** [reply]
It's mostly because core packages like accounts are tightly bound to Mongo so you can't run Meteor, for example fully with an sql db, even if there are already packages out there: you basically you end up with managing two databases. 

Second, I still don't like mongos sspl license move, even if this will not affect 99.9% of us. It's just not a real free license anymore.

 Mongo was one of the first js capable nosql DBS and a perfect choice back then but nowadays there are more of these.

I also agree this not an urgent issue but rather something to put into long-term perspective.

**Reply by Saeeed-B** [reply]
It's scary that so many packages, like accounts, are dependent on monogDb

**Comment by radekmie** [comment]
I'm 100% for it. But not because I think MongoDB is not the way to go (I'd most likely use it anyway), but because issues like https://github.com/meteor/meteor/issues/11382 simply wouldn't be there. It'd also make implementing a custom backend (e.g., Auth0) _much_ simpler.

As for the implementation, I'd see it in two parts:
1. Extract the DB operations into new functions (e.g., `Accounts._resolveUserByEmail`).
2. Remove the implementation from the packages and move it to `x-backend-mongo` (or similar) package.
